syntax = "proto3";

package ai.mantik.engine.protos;
option go_package = "mantik/engine";

import "google/protobuf/empty.proto";

/** Returns information about the Engine process. */
service AboutService {
    rpc Version (google.protobuf.Empty) returns (VersionResponse) {}

    // Request client configuration
    rpc ClientConfig(google.protobuf.Empty) returns (ClientConfigResponse){}
}

/**
Contains the current Mantik Engine version.
*/
message VersionResponse {
    string version = 1;
}

message ClientConfigResponse {
    // JSON Serialized Client Config
    string config = 1;
}